contracts/external/cw-bounties/src/contract.rs

```diff
@@ -0,0 +1,272 @@
+use std::cmp::Ordering;
+
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, BankMsg, Binary, Coin, Deps, DepsMut, Env, MessageInfo, Order, Response, StdResult,
+};
+use cw2::set_contract_version;
+use cw_paginate::paginate_map_values;
+use cw_utils::must_pay;
+
+use crate::{
+    error::ContractError,
+    msg::{ExecuteMsg, InstantiateMsg, QueryMsg},
+    state::{Bounty, BountyStatus, BOUNTIES, ID},
+};
+
+pub(crate) const CONTRACT_NAME: &str = "crates.io:cw-bounties";
+pub(crate) const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let owner = deps.api.addr_validate(&msg.owner)?;
+
+    // Set the contract owner
+    cw_ownable::initialize_owner(deps.storage, deps.api, Some(owner.as_str()))?;
+
+    // Initialize the next ID
+    ID.save(deps.storage, &1)?;
+
+    Ok(Response::default())
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    // Only the owner can execute messages on this contract
+    cw_ownable::assert_owner(deps.storage, &info.sender)?;
+
+    match msg {
+        ExecuteMsg::Close { id } => close(deps, info, id),
+        ExecuteMsg::Create {
+            amount,
+            title,
+            description,
+        } => create(deps, env, info, amount, title, description),
+        ExecuteMsg::PayOut { id, recipient } => pay_out(deps, env, id, recipient),
+        ExecuteMsg::Update {
+            id,
+            amount,
+            title,
+            description,
+        } => update(deps, env, info, id, amount, title, description),
+        ExecuteMsg::UpdateOwnership(action) => update_owner(deps, info, env, action),
+    }
+}
+
+pub fn create(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    amount: Coin,
+    title: String,
+    description: Option<String>,
+) -> Result<Response, ContractError> {
+    // Check funds sent match the bounty amount specified
+    let sent_amount = must_pay(&info, &amount.denom)?;
+    if sent_amount != amount.amount {
```

Comment:
@JakeHartnell quick question. Funds are created per bounty, right? Like 100 JUNO for Bounty 1, and 100 JUNO for Bounty 2. If this is so, when creating 2nd bounty it must check total funds on contract with sum of funds over all bounties, I guess.